File: rb/features.rb
# NpmRegistry SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NpmRegistryFeatures
  def self.make_feature(name)
    case name
    when "base"
      NpmRegistryBaseFeature.new
    when "test"
      NpmRegistryTestFeature.new
    else
      NpmRegistryBaseFeature.new
    end
  end
end

File: py/features.py
# NpmRegistry SDK feature factory

from feature.base_feature import NpmRegistryBaseFeature
from feature.test_feature import NpmRegistryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NpmRegistryBaseFeature(),
        "test": lambda: NpmRegistryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
